app/hooks/useTaco.ts

```diff
@@ -0,0 +1,101 @@
+import { SiweMessage } from "@didtools/cacao";
+import {
+  conditions,
+  decrypt,
+  encrypt,
+  ThresholdMessageKit,
+} from "@nucypher/taco";
+import {
+  SingleSignOnEIP4361AuthProvider,
+  USER_ADDRESS_PARAM_EXTERNAL_EIP4361,
+} from "@nucypher/taco-auth";
+import { DIDSession } from "did-session";
+import { ethers } from "ethers";
+import { useContext } from "react";
+import { TACoContext } from "../context/TACoContext";
+
+export default function useTaco() {
+  const { isInitialized, ritualId, domain } = useContext(TACoContext);
+
+  async function encryptWithTACo(
+    messageToEncrypt: string,
+    condition: conditions.condition.Condition,
+    provider: ethers.providers.Provider,
+    signer: ethers.Signer,
+  ) {
+    if (!isInitialized) return;
+
+    const tmk = await encrypt(
+      provider,
+      domain,
+      messageToEncrypt,
+      condition,
+      ritualId,
+      signer,
+    );
+
+    return encodeB64(tmk.toBytes());
+  }
+
+  async function decryptWithTACo(
+    encryptedMessage: string,
+    provider: ethers.providers.Provider,
+  ) {
+    if (!isInitialized) return;
+
+    const siweInfo = await loadSiweFromOrbisSession();
+    if (!siweInfo.message || !siweInfo.signature) {
+      console.error("No SIWE message or signature missing");
+      return;
```

Comment:
Much like `"<Decryption failed>"` below should this scenario also return a non-empty failure string, or is that handled separately?

Reply:
Yes! Done in c05ab54f6efed85b4af612829f0aa7e30b31b0fa